tests/ui/unnecessary_to_owned_on_split.rs

```diff
@@ -0,0 +1,6 @@
+fn main() {
+    let _ = "a".to_string().split('a').next().unwrap();
+    //~^ ERROR: unnecessary use of `to_string`
+    let _ = "a".to_owned().split('a').next().unwrap();
```

Comment:
I'd like to see a test splitting by a &str and perhaps a counter-test that calls split by `()` which is implemented by a custom trait (if that is even possible?).

Reply:
Implementing the `Pattern` trait is... quite a lot of code. So instead, I added checks for both `char` and `str` and also added the equivalent tests on `slice`.